### PR TITLE
Resources Test Case

### PR DESCRIPTION
**Pull request by jorgeorpinel**

### CODE REVIEW

File: test/Controllers/ResourcesTest.php [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        $this->assertCount($perPage, $response->resources); // Slightly hardcoded to assume $resources exists in $response.
+        foreach($response->resources as $resource) {
+            $this->assertInstanceOf('GonebusyLib\Models\GetResourcesResponse', $response);
+        }
+    }
+
+    /**
+     * Test GET /resources/things
+     * XXX Assumes there's 0 Things in the API's database.
+     * Test GonebusyLib\Controllers\ResourcesController::getResourceThings()
```
Comment (jorgeorpinel) [inline test/Controllers/ResourcesTest.php]: The only Q related to this PR @alexagranov, is that there's no things coming from the API so this test isn't gonna do much for now, in fact it's hardcoded to expect to receive an empty array.

File: test/Controllers/ResourcesTest.php [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+     * @param  string $type Should be 'create' or 'update'.
+     * @return  CreateResourceBody or UpdateResourceByIdBody object with unique data to send to API
+     */
+    private function resourceData($type) {
+        $rand = rand();
+        switch($type) {
+            case 'create':
+                return new CreateResourceBody(
+                    "name",
+                    "Staff", // Hardcoded. Way to get from API or SDK in the future?
```
Comment (alexagranov) [inline test/Controllers/ResourcesTest.php]: Can we remove this comment?  Types will just come from API documentation.

Comment (jorgeorpinel) [inline test/Controllers/ResourcesTest.php]: Removed.

**Comment by jorgeorpinel** [comment]
> For Things, I'll make a note to add some Things in Sandbox to test with in next day or two.

OK, thanks. Lmk

File: test/Controllers/ResourcesTest.php [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    private function resourceBody($type) {
+        $rand = rand();
+        switch($type) {
+            case 'create':
+                return new CreateResourceBody(
+                    "name",
+                    "Staff",
+                    NULL, // capacity
+                    "description",
+                    NULL, // gender
```
Comment (alexagranov) [inline test/Controllers/ResourcesTest.php]: Let's go ahead and pass a gender: `[ F | M | T/G ]`

**Comment by alexagranov** [comment]
@jorgeorpinel please resolve conflicts to continue.

File: test/Controllers/ResourcesTest.php (hunk 3 above)
Comment (jorgeorpinel) [inline test/Controllers/ResourcesTest.php]: Done. But there may be a problem with the API...

File: test/Controllers/ResourcesTest.php [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+                    NULL // $response->resource->ownerId
+                );
+            case 'update':
+                return new UpdateResourceByIdBody(
+                    $response->resource->capacity,
+                    $response->resource->description,
+                    GenderEnum::M, //$response->resource->gender, // XXX Hardcoded as API isn't returning a value
+                    $response->resource->name,
+                    $response->resource->thingTypeId
+                );
```
Comment (jorgeorpinel) [inline test/Controllers/ResourcesTest.php]: @alexagranov if you see the `XXX` codetags above, this means the API isn't returning `gender` neither after a POST /resources/new nor PUT /resources/{id}, not even on GET /resources/{id} so I had to hardcode the mocked response to match the data I sent in the first place.

image

image

**Comment by jorgeorpinel** [comment]
@alexagranov conflicts fixed.

File: test/Controllers/ResourcesTest.php (hunk 4 above)
Comment (alexagranov) [inline test/Controllers/ResourcesTest.php]: Good catch.  I'll log a bug.

**Comment by jorgeorpinel** [comment]
Good morning @alexagranov . Are you still requesting changes in this PR? BTW I removed the `XXX` comments since both problems have been logged here: API not returning gender and no Things existing in the system. For the latter, also, I changed the `getResourceThings()` test so it actually expects to find at least 3 things -- so PHPUnit will fail until the API returns them but if you review the Travis job logs there's only that one "error":
```sh
There was 1 failure:

1) ResourcesTest::testGetResourceThings
Failed asserting that actual size 0 matches expected size 3.

/home/travis/build/gonebusy/gonebusy-php-client/test/Controllers/ResourcesTest.php:207
```

**Comment by jorgeorpinel** [comment]
p.s. Wow. It takes a while for CI to run `phpunit` now that several test cases are merged into your master... Wishing for Prism right now.

**Comment by alexagranov** [comment]
Ok @jorgeorpinel some ThingTypes exist in Sandbox now:
```
curl -X GET --header 'Accept: application/json' --header 'Authorization: Token ac98ed08b5b0a9e7c43a233aeba841ce' 'https://sandbox.gonebusy.com/api/v1/resources/things?page=1&per_page=10'
```

Don't hate on Travis, he's a gentlemen and a scholar.  And free ;-)

**Comment by jorgeorpinel** [comment]
Haha it's not Travis fault, I know. Just a lot of API calls to run.

So the build should pass now. How do you ask Travis to run the same job again? Hm...

**Comment by jorgeorpinel** [comment]
@alexagranov I just updated to 0.0.9 and pushed this up so Travis ran, passing now a expected (:

**Comment by alexagranov** [comment]
@jorgeorpinel #16 was merged 5 days ago :-)  So yes it was first.  I think I'll go ahead and merge this so we can keep moving.

**Pull request merged**